interfaces/apparmor/export_test.go

```diff
@@ -20,9 +20,71 @@
 package apparmor
 
 import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
 	"github.com/snapcore/snapd/testutil"
 )
 
+var (
+	IsHomeUsingNFS                  = isHomeUsingNFS
+	SetupSnapConfineGeneratedPolicy = setupSnapConfineGeneratedPolicy
+)
+
+// MockSetupSnapConfineGeneratedPolicy mocks the setup of snap-confine policy.
+func MockSetupSnapConfineGeneratedPolicy(f func() error) (restore func()) {
+	old := setupSnapConfineGeneratedPolicy
+	setupSnapConfineGeneratedPolicy = f
+	return func() {
+		setupSnapConfineGeneratedPolicy = old
+	}
+}
+
+//MockMountInfo mocks content of /proc/self/mountinfo read by isHomeUsingNFS
+func MockMountInfo(text string) (restore func()) {
+	old := procSelfMountInfo
+	f, err := ioutil.TempFile("", "mountinfo")
+	if err != nil {
+		panic(fmt.Errorf("cannot open temporary file %s", err))
+	}
+	if err := ioutil.WriteFile(f.Name(), []byte(text), 0644); err != nil {
+		panic(fmt.Errorf("cannot write mock mountinfo file %s", err))
+	}
+	procSelfMountInfo = f.Name()
+	return func() {
+		os.Remove(procSelfMountInfo)
+		procSelfMountInfo = old
+	}
+}
+
+// MockEtcFstab mocks content of /etc/fstab read by isHomeUsingNFS
+func MockEtcFstab(text string) (restore func()) {
+	old := procSelfMountInfo
+	f, err := ioutil.TempFile("", "fstab")
+	if err != nil {
+		panic(fmt.Errorf("cannot open temporary file %s", err))
+	}
+	if err := ioutil.WriteFile(f.Name(), []byte(text), 0644); err != nil {
+		panic(fmt.Errorf("cannot write mock fstab file %s", err))
+	}
+	etcFstab = f.Name()
+	return func() {
+		os.Remove(etcFstab)
```

Comment:
We need a check here that panics if this is /etc/fstab. We don't want to kill the system by mistake.

Reply:
Oh, good idea. Done